lib/phoenix_live_view.ex

```diff
@@ -532,6 +532,39 @@ defmodule Phoenix.LiveView do
     end
   end
 
+  @doc """
+  Returns true if the socket is connected.
+
+  `is_connected?/1` is a guard where `connected?/1` is a function.
+
+  `is_connected?/1` enables splitting `mount/3` callbacks into connected and disconnected function heads:
+
+  ```elixir
+  defmodule DemoWeb.ClockLive do
+    use Phoenix.LiveView
+
+    def mount(_params, _session, socket) when is_connected?(socket) do
+      # Socket is connected, perform stateful work
+      :timer.send_interval(1000, self(), :tick)
+
+      {:ok, assign(socket, date: :calendar.local_time())}
+    end
+
+    def mount(_params, _session, socket) do
+      # Socket is disconnected
+      {:ok, socket}
+    end
+
+    def handle_info(:tick, socket) do
+      {:noreply, assign(socket, date: :calendar.local_time())}
+    end
+  end
+  ```
+
+  Like `connected?/1`, use `is_connected?/1` to conditionally perform stateful work, such as subscribing to PubSub topics, sending messages, etc.
+  """
+  defguard is_connected?(socket) when socket.transport_id != nil
```

Comment:
Guards do not have a question mark:

```suggestion
  defguard is_connected(socket) when socket.transport_id != nil
```

I am fine with adding this but we should probably deprecate `connected?`. Thoughts @chrismccord?